manifests/service.pp

```diff
@@ -143,6 +143,9 @@
   elsif($os =~ /opensuse/ and $release == '13') {
     $service_provider = 'systemd'
   }
+  elsif($os =~ /amazon/ and $release > '2') {
```

Comment:
This comparison [fails](https://travis-ci.org/elastic/puppet-logstash/jobs/385371176) the puppet-lint "version comparison check".

Please have a look here for a description and advice on how to resolve:

https://github.com/voxpupuli/puppet-lint-version_comparison-check#checks

Reply:
Amazon Linux 2 is currently at LTS candidate, so it's not 100% official yet....

As for how it interprets the $release variable, Amazon Linux version 2 sets it as 2, while the previous versions of Amazon Linux sets it as the year the image is released... 2018, 2017, and so forth.  So, chornologically, it goes from 2012, 2013..., 2018, then to 2.   That's the reason for the  weird condition.  I'm fairly sure that, with the systemd based Amazon Linux version 2, puppet will figure out the provider correctly; it's just these previous versions that cannot.

I will also take look at the version check and see what's wrong... I tried to follow the previous if/else conditions, and the exact change that worked fine on my puppet setup.  From what I see, I have set this as a string.